src/routes/root/Summary.tsx

```diff
@@ -1,73 +1,147 @@
 import { useNavigate } from 'react-router-dom';
 import Contract from 'web3-eth-contract';
-import { useState } from 'react';
+import { useEffect, useState } from 'react';
 import { Heading } from '../../components/Heading';
 import { Button } from '../../components/Button';
 import { useEnsuredWallet } from '../../context/WalletContext';
-import { DemoAbi as Abi } from '../../abi';
-import { getNetwork, Networks } from '../../utils/network';
+import { Networks, getNetwork } from '../../utils/network';
+import { DepositAdapterABI, DEPOSIT_ADAPTER_ORIGIN } from '../../contracts';
+import { DepositDataJSON } from '../../components/JSONDropzone/validation';
+import { useStorage } from '../../context/StorageContext';
+
+// mocks
+const value = '{value}';
+const currency = '{currency}';
 
 export function Summary(): JSX.Element {
   const wallet = useEnsuredWallet();
-  const [network, setNetwork] = useState<Networks | null>(null);
+  const storage = useStorage();
+  const navigate = useNavigate();
+
+  const [selectedNetwork, setSelectedNetwork] = useState<Networks | null>(null);
+  const [manuallyChaingedNetwork, setManuallyChaingedNetwork] = useState<string | null>(
+    null,
+  );
+
   const [errorMsg, setErrorMsg] = useState<string>();
 
-  // mocks
-  const value = '{value}';
-  const currency = '{currency}';
+  useEffect(() => {
+    if (wallet.web3.currentProvider) {
+      // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+      const provider = wallet.web3.currentProvider as any;
+      // eslint-disable-next-line @typescript-eslint/no-unsafe-call, @typescript-eslint/no-unsafe-member-access
+      provider.on('chainChanged', (chainId: number) => {
```

Comment:
@BeroBurny Could you check whats happening here. I can see web3 has chainChanged event types in Web3BaseProvider. Could you figure out why this callback is different from typed one

Reply:
https://github.com/web3/web3.js/issues/5939 :eyes: